python/scripts/import_dicom_study.py

```diff
@@ -0,0 +1,314 @@
+        "\t                  'database_config.py')\n"
+        "\t-s, --source    : Path of the source directory containing the DICOM files of the"
+        "\t                  study.\n"
+        "\t    --overwrite : Overwrite the DICOM archive file if it already exists.\n"
+        "\t    --insert    : Insert the created DICOM archive in the database (requires the archive\n"
+        "\t                  to not be already inserted).\n"
+        "\t    --update    : Update the DICOM archive in the database (requires the archive to be\n"
+        "\t                  already be inserted), generally used with '--overwrite'.\n"
+        "\t    --session   : Associate the DICOM study with an existing session using the LORIS-MRI\n"
+        "\t                  Python configuration.\n"
+        "\t-v, --verbose   : If set, be verbose\n"
+        "\n"
+        "Required options: \n"
+        "\t--profile\n"
+        "\t--source\n"
+        "\t--target\n"
+    )
+
+    # NOTE: Some options do not have short options but LorisGetOpt does not support that, so we
+    # repeat the long names.
+    options_dict = {
+        "profile": {
+            "value": None, "required": True, "expect_arg": True, "short_opt": "p", "is_path": False
+        },
+        "source": {
+            "value": None, "required": True,  "expect_arg": True, "short_opt": "s", "is_path": True,
+        },
+        "overwrite": {
+            "value": False, "required": False, "expect_arg": False, "short_opt": "overwrite", "is_path": False,
+        },
+        "insert": {
+            "value": False, "required": False, "expect_arg": False, "short_opt": "insert", "is_path": False,
+        },
+        "update": {
+            "value": False, "required": False, "expect_arg": False, "short_opt": "update", "is_path": False,
+        },
+        "session": {
+            "value": False, "required": False, "expect_arg": False, "short_opt": "session", "is_path": False,
+        },
+        "verbose": {
+            "value": False, "required": False, "expect_arg": False, "short_opt": "v", "is_path": False
+        },
+        "help": {
+            "value": False, "required": False, "expect_arg": False, "short_opt": "h", "is_path": False
+        },
+    }
+
+    # Get the CLI arguments and connect to the database.
+
+    loris_getopt_obj = LorisGetOpt(usage, options_dict, os.path.basename(__file__[:-3]))
+    env = make_env(loris_getopt_obj)
+    args = Args(loris_getopt_obj.options_dict)
+
+    # Check arguments.
+
+    if not os.path.isdir(args.source) or not os.access(args.source, os.R_OK):
+        log_error_exit(
+            env,
+            "Argument '--source' must be a readable directory path.",
+            lib.exitcode.INVALID_ARG,
+        )
+
+    if args.insert and args.update:
+        log_error_exit(
+            env,
+            "Arguments '--insert' and '--update' cannot be used both at the same time.",
+            lib.exitcode.INVALID_ARG,
```

Comment:
Why make the distinction between `--insert` and `--update` here? I think a simple --database option is enough.

If --overwrite is used, then, it would update, if not, it would insert after having check that there is not already an entry in the DB for the StudyUID.

Reply:
ok. Interesting. I have to think about this one. I would have a tendency to have a simple database flag and then figure out if data should be updated or inserted in the database when want to update the database instead of having to do that outside of the script in order to properly call import_dicom_study.py... I will sleep on it and get back to you.